docs/enhancements/on-prem-data-collection.md

```diff
@@ -0,0 +1,220 @@
+
+#### Risk 3: DDoS-ing Insights Ingress Server
+
+Without any constraints, it's possible for the assisted service to send data to the data collection server repeatedly. This can prevent other clients from sending their data, and cause loss of data, or prevent us from reaching the data since the server will be busy with the repeated connections.
+
+##### Mitigation
+
+1. Clearly defining when events are sent (see `Data Sending Flow` under the `Proposal` section above)
+2. Data is sent as best effort
+3. Retry will be implemented as a future enhancement
+    - Not retrying until a timeout has been reached and there's no response from the server
+    - Each retry timeout will be longer (exponential backoff and jitter)
+    - Data will no longer be sent for this cluster once the maximum number of retries is hit
+4. Using a remote configuration (see section under `Proposal` above)
+
+#### Risk 4: Duplicate Data
+
+With all the cases for sending data, there's a high chance for duplicated data to be sent to the data collection server. It can skew the analysis of data and it won't be representative of the actual data from the field.
+
+##### Mitigation
+
+A field can be used as a marker to indicate the data has been sent to the event streaming service or that the data already exists. One option is to use the timestamp, cluster ID, and user email to detect duplicate events.
+
+## Design Details
+
+### Open Questions
+
+1. Is the remote configuration a good idea? A good amount of customers using this feature are Telcos and they would be disconnected
+2. What is the Telemetry opt-out? Is it the pull secret not containing the auth for it? Would this be relevant for disconnected? Wouldn't it be better to use a specific opt-out variable to not run this service of pushing data to a server (local or remote)
+3. How would the configuration for writing to a local s3 storage work? Would it be something we define and tell the customer to set up? Would it be something the customer defines and set up? Ideally would the customer need to be involved at all? Wouldn't there be a big space requirement for this?
+4. How are the variable names? Are there conventions/opinions?
+
+### UI Impact
+
+No UI changes will need to be made.
+
+### Test Plan
+
+#### Test 1
+
+Add a check at the end of a cluster install attempt that will query the correct data made it to the data hosting service if the feature is enabled.
+
+#### Test 2
+
+Check against a large-scale cluster with > 100 hosts to verify:
+1. All of the cluster's events are being stored, queried, and sent (within a reasonable time)
+2. That the event streaming can receive and process the large data set
+
+## Drawbacks
+
+- Dependency on the data hosting service selected in [MGMT-11244](https://issues.redhat.com/browse/MGMT-11244) could cause delays in this epic being completed.
+- Disconnected environment users would still need to send data to us
```

Comment:
I don't think this is worth doing. I feel like it's incredibly unlikely that a customer goes out of their way to send us telemetry manually that isn't going to benefit them directly (solving a support issue or something).

If this involves no extra work then I guess it's fine to develop some kind of process for doing this, but I don't think we should design our code changes around this being something that might actually happen.

Reply:
That's right! The meeting helped clarified the end goal so this is not relevant at all. Thanks!